local/network.go

```diff
@@ -127,25 +128,58 @@ func init() {
 	}
 }
 
+// NodeProcessCreator is an interface for new node process creation
 type NodeProcessCreator interface {
 	NewNodeProcess(config node.Config, args ...string) (NodeProcess, error)
 }
 
-type nodeProcessCreator struct{}
+type nodeProcessCreator struct {
+	color logging.Color
+}
+
+// default variables pointing to the default functions to get the os output files
+// we can replace these in testing (see TestChildCmdRedirection in local/network_test.go)
+var (
+	stdOutFunc = getDefaultStdOut
+	stdErrFunc = getDefaultStdErr
+)
```

Comment:
I think we can avoid having the globals/overwriting them in tests by adding additional fields to `nodeProcessCreator`.

Namely, we can add fields `stdout io.Writer` and `stderr io.Writer` to control where stdout and stderr are written for testing

Reply:
Much better and elegant, also uses way less code